packages/scanner/src/report/scanSummary.ts

```diff
@@ -20,3 +20,29 @@ export interface ScanSummary {
   numFindings: number;
   appMapMetadata: AppMapMetadata;
 }
+
+export function aggregateScanSummary(
+  sourceScanSummary: ScanSummary,
+  targetScanSummary: ScanSummary
+) {
+  targetScanSummary.numAppMaps += sourceScanSummary.numAppMaps;
+  targetScanSummary.numChecks += sourceScanSummary.numChecks;
+  targetScanSummary.rules = [...new Set(targetScanSummary.rules.concat(sourceScanSummary.rules))];
+  targetScanSummary.ruleLabels = [
+    ...new Set(targetScanSummary.ruleLabels.concat(sourceScanSummary.ruleLabels)),
+  ];
+  targetScanSummary.numFindings += sourceScanSummary.numFindings;
+
+  // we don't need the metadata.
+  // appMapMetadata.Git may also contain secrets we don't want to transmit.
+  targetScanSummary.appMapMetadata.labels = [];
+  targetScanSummary.appMapMetadata.apps = [];
+  targetScanSummary.appMapMetadata.clients = [];
+  targetScanSummary.appMapMetadata.frameworks = [];
+  targetScanSummary.appMapMetadata.git = [];
+  targetScanSummary.appMapMetadata.languages = [];
+  targetScanSummary.appMapMetadata.git = [];
+  targetScanSummary.appMapMetadata.recorders = [];
+  targetScanSummary.appMapMetadata.testStatuses = [];
+  targetScanSummary.appMapMetadata.exceptions = [];
```

Comment:
re: default constructor arguments - on the other hand, this looks to me like this isn't the ideal type to be aggregating with.

Reply:
Now aggregating on ScanResults.